src/platform_impl/windows/event_loop.rs

```diff
@@ -223,67 +223,60 @@ impl<T: 'static> EventLoop<T> {
                 runner.new_events();
                 loop {
                     if !unread_message_exists {
-                        if 0 == winuser::PeekMessageW(&mut msg, ptr::null_mut(), 0, 0, 1) {
+                        if 0 == winuser::PeekMessageW(
+                            &mut msg,
+                            ptr::null_mut(),
+                            0,
+                            0,
+                            winuser::PM_REMOVE,
+                        ) {
                             break;
                         }
                     }
-                    if msg.message == winuser::WM_PAINT {
-                        unread_message_exists = true;
-                        break;
-                    }
                     winuser::TranslateMessage(&mut msg);
                     winuser::DispatchMessageW(&mut msg);
 
                     unread_message_exists = false;
+
+                    if msg.message == winuser::WM_PAINT {
+                        break;
+                    }
```

Comment:
Now that this `WM_PAINT` check moved after `TranslateMessage` and `DispatchMessage`, the first `WM_PAINT` event will be handled before `main_events_cleared` is called. This won't cause any bugs because the runner automatically dispatches `MainEventsCleared` before `RedrawRequested`, but this code still looks like it should be buggy upon first glance. Could you add a comment explaining why it works?

Reply:
done